docs/docsite/rst/reference_appendices/general_precedence.rst

```diff
@@ -0,0 +1,131 @@
+
+.. contents::
+   :local:
+
+Precedence categories
+---------------------
+
+Ansible supports four sources for controlling its behavior. In order of precedence from lowest (most easily overridden) to highest (overrides all others), the categories are:
+
+   Configuration settings
+   Command-line options
+   Playbook keywords
+   Variables
+
+Each category overrides any information from all lower-precedence categories. For example, a playbook keyword will override any configuration setting.
+
+Within each precedence category, specific rules apply. However, generally speaking, 'last defined' wins and overrides any previous definitions.
+
+Configuration settings
+^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Configuration settings<ansible_configuration_settings>` include both values from the ``ansible.cfg`` file and environment variables. Within this category, values set in configuration files have lower precedence. Ansible uses the first ``ansible.cfg`` file it finds, looking in these locations in order:
+
+   #. the file specified by the value of ``ANSIBLE_CONFIG``
+   #. config in the current working directory: `./ansible.cfg`
+   #. config in the current user's home directory: `~/.ansible.cfg`
+   #. global config: `/etc/ansible/ansible.cfg`
+
+Environment variables have a higher precedence than entries in ``ansible.cfg``. If you have environment variables set, they override the settings in whichever ``ansible.cfg`` file Ansible loads. The value of any given environment variable follows normal shell precedence: the last value defined overwrites previous values. (TODO: Does Ansible use the envvars defined on the control node or on the managed node?)
+
+You can use the :ref:`ansible-config` command line utility to see the current value of a configuration item and where it came from. (TODO: How does this work? Add example.)
+
+Command-line options
+^^^^^^^^^^^^^^^^^^^^
+
+Any command-line option will override any configuration setting.
+
+At the command line, if you pass multiple values for a parameter that accepts only a single value, the last defined value wins. For example, this :ref:`ad-hoc task<intro_adhoc>` will connect as ``carol``, not as ``mike``::
+
+      ansible -u mike -m ping myhost -u carol
+
+Some parameters allow multiple values. In this case, Ansible will append all values from the hosts listed in inventory files inventory1 and inventory2::
+
+   ansible -i /path/inventory1 -i /path/inventory2 -m ping all
+
+The help for each :ref:`command-line tool<command_line_tools>` specifies which parameters allow for multiple entries.
```

Comment:
Hm. Looks like this is not exactly true: https://docs.ansible.com/ansible/devel/cli/ansible.html - I don't see anything about "allows multiple entries" here.

Reply:
addressed by f3ce501e720e2ba577873a22dc641fe470f4444d